app/src/main/java/net/sourceforge/opencamera/ui/MainUI.java

```diff
@@ -1355,6 +1375,13 @@ public void updateExposureLockIcon() {
         view.setContentDescription( main_activity.getResources().getString(enabled ? R.string.exposure_unlock : R.string.exposure_lock) );
     }
 
```

Comment:
I have seen some setText in controllers before - it will be better to move all work with views to this class

Reply:
A `TextView` with its `setText` is used in `SoftwareSyncController` as a container for a string (the text itself) and its color, which is actually displayed to user by `DrawPreview` [here](https://github.com/MobileRoboticsSkoltech/OpenCamera-Sensors/blob/2dc0af7927a5e324940a510610c715ab98aa7ccc/app/src/main/java/net/sourceforge/opencamera/ui/DrawPreview.java#L1242). I don't think that it is a good idea to create the string here, because its contents may vary a lot, but we can replace `TextView` with a string its color separated - is it a better approach?

Reply:
Yes, main problem is that it is dangerous to use refs to views from services which can possibly live more than activity or view - you can lose memory